### PR TITLE
docs: hooks docs and gradientheadline fix

### LINKED ISSUE #1742: Document `@marigold/system`
## Description

Currently there is no documentation for your `system` package, expect `Box` and `SVG`.

## Context

We need to have documentation about the "inner workings" too, especially because `useTheme` and `useResponsiveValues` are hooks from this package.

### PR DESCRIPTION
**Pull request by sarahgm**
**included**:
- useResponsiveValue
- useTheme
- useAsyncDataList
- SVG

closes #1742

### CODE REVIEW

File: docs/src/pages/components/useAsyncList.mdx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+---
```
Comment (sebald) [inline docs/src/pages/components/useAsyncList.mdx:1]: /hooks instead of /components? :)

File: docs/src/pages/components/useTheme.mdx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+github: https://github.com/marigold-ui/marigold/blob/main/packages/system/src/hooks/useTheme.tsx
+edit: https://github.com/marigold-ui/marigold/blob/main/docs/src/pages/components/useTheme.mdx
+switchTheme: true
+---
+
+The `useTheme` is a hook that gets you the current theme and all of its attributes and properties.
+
+The `useTheme` hook returns tree parameters:
+
+- **First:** returns the current `theme`, if there isn't a theme it uses the `__defaultThme`
```
Comment (sebald) [inline docs/src/pages/components/useTheme.mdx:14]:
```suggestion
- **First:** returns the current `theme`, if there isn't a theme it uses the `__defaultTheme`
```

Comment (sebald) [inline docs/src/pages/components/useTheme.mdx:14]: Props Table and list`theme``css` and `get` ? :)

**Review by sebald — CHANGES_REQUESTED** [review]

**Review by benediktgrether — DISMISSED** [review]

File: docs/src/pages/components/useTheme.mdx (hunk 2 above)
Comment (sarahgm) [reply docs/src/pages/components/useTheme.mdx:14]: but are they props I thought its just the returns

Comment (sebald) [reply docs/src/pages/components/useTheme.mdx]: Yes they are 🤔 can we still use the same table somehow?

Comment (sarahgm) [reply docs/src/pages/components/useTheme.mdx]: I think no, because the PropsTable has a hard-coded header 🤔

**Review by sebald — APPROVED** [review]

**Pull request merged**